code/game/machinery/doors/firedoor.dm

```diff
@@ -233,7 +233,7 @@ var/global/list/alert_overlays_global = list()
 			update_icon()
 			return
 
-	if( iscrowbar(C) || ( istype(C,/obj/item/weapon/fireaxe) && C.wielded ) )
+	if( iscrowbar(C) || isEmag(C) || ( istype(C,/obj/item/weapon/fireaxe) && C.wielded ) )
```

Comment:
Why is this before the welded check? An emag can't do anything to a welded door.